poc_mobile_verifier/src/verifier.rs

```diff
@@ -107,18 +114,27 @@ impl VerifierDaemon {
 
     pub async fn reward_epoch(&mut self, epoch: Range<DateTime<Utc>>) -> Result {
         let heartbeats = Heartbeats::validated(&self.pool, epoch.start).await?;
+        let speedtests = SpeedtestAverages::validated(&self.pool, epoch.start).await?;
 
-        let rewards = self.verifier.reward_epoch(&epoch, heartbeats).await?;
+        let rewards = self
+            .verifier
+            .reward_epoch(&epoch, heartbeats, speedtests)
+            .await?;
 
         let mut transaction = self.pool.begin().await?;
 
-        // Clear the heartbeats database
+        // Clear the heartbeats and speedtest tables:
         // TODO: should the truncation be bound to a given epoch?
         // It's not intended that any heartbeats will exists outside the
         // current epoch, but it might be better to code defensively.
-        sqlx::query("TRUNCATE TABLE heartbeats;")
-            .execute(&mut transaction)
-            .await?;
+        sqlx::query(
+            r#"
+            TRUNCATE TABLE heartbeats;
+            TRUNCATE TABLE speedtests;
```

Comment:
We can't truncate speedtests at the end of the rewardable period.  The rolling average can contain speedtests that cross the boundary.  

So the first speedtest received at the start of a new rewardable period will contain 5 from the previous period and 1 from the new period.